src/__integration_tests__/utils/helpers.js

```diff
@@ -53,6 +53,40 @@ export const authorizePaymentAndGetError = async (
   return await errorMessageElem.evaluate((el) => el.textContent);
 };
 
+/*
+* utility function to emulate edit psp list and sort. This function has many timeout to allow view transitions to end before try to click CTA buttons
+*/
+export const checkPspList = async (noticeCode, fiscalCode, email, cardData) => {
+  const pspEditButtonSelector = "#pspEdit";
+  const pspFeeSortButtonId = "#sortByFee";
+
+  await fillAndSubmitCardDataForm(noticeCode, fiscalCode, email, cardData);
+
+  const pspEditButton = await page.waitForSelector(pspEditButtonSelector);
+  await pspEditButton.click();
+  await page.waitForTimeout(1000);
```

Comment:
Don't use waitForTimeout because is [deprecated](https://github.com/cloudflare/puppeteer/blob/main/docs/api/puppeteer.page.waitfortimeout.md).
Btw don't use fixed timeout during test because you may be wasting time unnecessarily. 
For example waitForSelector wait only time is necessary until element appear.

Reply:
this is not a mistake, you are missing one info. 😄 

we needed waitForTimeout with explicit time because we need to wait for a drawer menu animation to properly complete in order to execute this test correctly.

Reply:
In this test we don't need animation, so you can disable it by adding this code on beforeAll test function:

```
page.on("load", () => {
    const content = `* {transition-duration: 0s !important;}`;
    page.addStyleTag({ content });
});
```

And then you could change test like this:

```
export const checkPspList = async (noticeCode, fiscalCode, email, cardData) => {
  const pspEditButtonSelector = "#pspEdit";
  const pspFeeSortButtonId = "#sortByFee";

  await fillAndSubmitCardDataForm(noticeCode, fiscalCode, email, cardData);
  const pspEditButton = await page.waitForSelector(pspEditButtonSelector);
  //open sidebar
  await pspEditButton.click();
  const pspFeeSortButton = await page.waitForSelector(pspFeeSortButtonId);
  await pspFeeSortButton.click();
  // get the list of divs
  const pspElements = await page.$$(".pspFeeValue");
  // Extract numeric content from each div and return as an array
  const numericContents = await Promise.all(
    Array.from(pspElements).map(async (element) => {
      const text = await element.evaluate((el) => el.textContent);
      // We want to skip the Dollar, Euro, or any currency placeholder
      let numbers = text.match(/[\d,]+/g); // This will match sequences of digits and commas
      let result = numbers ? numbers.join("").replace(",",".") : ""; // Join the matched numbers if any and replace , as separator with .
      return parseFloat(result) || 0; // Convert to number, default to 0 if NaN
    })
  );
  const closePspListButton = await page.waitForSelector("#closePspList");
  await closePspListButton.click();
  // wait for sidebar closure
  await page.waitForSelector(pspFeeSortButtonId, {hidden: true});
  return numericContents;
};
```

With this change you can speed up test execution

Reply:
leaving the animation makes the test true to the real FE as-is. 
The test should not alter the behavior of the FE to become quicker or easier written.

I think we should not alter the behavior of the FE in any way.
(also the timeouts are a few seconds, wich i consider acceptable on a pipeline that anyway takes more than 15min to complete)

Reply:
If you prefer to don't disable animations please remove deprecated method usage. 
![image](https://github.com/user-attachments/assets/47dbd455-7691-4409-aa13-24e9cadabfd5)

Also i suppose there is too much timeout because you need only on open and on close action.
[Here](https://github.com/pagopa/pagopa-checkout-tests/blob/257761a09d8550c53d59af1fa1a353d42e794681/e2e-tests/src/npg/psp-sorting.npg.integration.test.ts) is used less timeout for similar test

Reply:
ok, if you insist i will replace with promise method, but i will not reduce timeout value since we did multiple tests and we feel comfortable with the current values.

again: we had an internal discussion and decided that few seconds of delay are not an issue